datastream-utils/src/main/java/com/linkedin/datastream/common/zk/ZkClient.java

```diff
@@ -212,80 +155,6 @@ public String ensureReadData(final String path) {
     return ensureReadData(path, _zkSessionTimeoutMs);
   }
 
-  @Override
-  @SuppressWarnings("unchecked")
-  protected <T extends Object> T readData(final String path, final Stat stat, final boolean watch) {
-    long startT = System.nanoTime();
-    try {
-      byte[] data = retryUntilConnected(() -> _connection.readData(path, stat, watch));
-      return (T) deserialize(data);
-    } finally {
-      long endT = System.nanoTime();
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("readData, path: {}, time: {} ns", path, (endT - startT));
-      }
-    }
-  }
-
-  @Override
-  public void writeData(final String path, Object data, final int expectedVersion) {
-    long startT = System.nanoTime();
-    try {
-      final byte[] bytes = serialize(data);
-
-      retryUntilConnected(() -> {
-        _connection.writeData(path, bytes, expectedVersion);
-        return null;
-      });
-    } finally {
-      long endT = System.nanoTime();
-      if (LOG.isTraceEnabled()) {
-        LOG.trace("writeData, path: {}, time: {} ns", path, (endT - startT));
-      }
-    }
-  }
-
-  @Override
-  public String create(final String path, Object data, final CreateMode mode) throws RuntimeException {
-    if (path == null) {
-      throw new IllegalArgumentException("path must not be null.");
```

Comment:
The new library throws `NullPointerException` instead of `IllegalArgumentException`. Can you plz check the contract of new methods ? Mostly the unit-tests are not covering it, otherwise the tests should have failed. May be a good idea to add those tests

Reply:
Also, this is something to bring up with Helix team that their contact says `IllegalArgumentException` but internally the method throws `NullPointerException`.

Reply:
Added unit test for NullPointerException. Will bring up the discrepancy with Helix team separately